docs/user-guides/server-side-apply.md

```diff
@@ -0,0 +1,234 @@
+
+```yaml
+managedFields:
+- apiVersion: monitoring.rhobs/v1
+  fieldsType: FieldsV1
+  fieldsV1:
+    f:metadata:
+      f:labels:
+        f:app.kubernetes.io/managed-by: {}
+        f:app.kubernetes.io/name: {}
+        f:app.kubernetes.io/part-of: {}
+    f:spec:
+      f:enforcedSampleLimit: {}
+  manager: kubectl
+  operation: Apply
+```
+
+ ## Modifying a field managed by MonitoringStack
+
+ - Change `spec.LogLevel`, a field managed by MonitoringStack using the following yaml and command
+
+```yaml
+$ cat prom-stack-edited.yaml 
+# changing the logLevel from debug to info
+apiVersion: monitoring.rhobs/v1
+kind: Prometheus
+metadata:
+  labels:
+    app.kubernetes.io/managed-by: observability-operator
+    app.kubernetes.io/name: sample-monitoring-stack
+    app.kubernetes.io/part-of: sample-monitoring-stack
+  name: sample-monitoring-stack
+  namespace: obo-demo
+spec:
+  logLevel: info
+
+
+$ oc apply -f ./prom-spec-edited.yaml --server-side 
+error: Apply failed with 1 conflict: conflict with "obo-demo/sample-monitoring-stack": .spec.logLevel
+Please review the fields above--they currently have other managers. Here
+are the ways you can resolve this warning:
+* If you intend to manage all of these fields, please re-run the apply
+  command with the `--force-conflicts` flag.
+* If you do not intend to manage all of the fields, please edit your
+  manifest to remove references to the fields that should keep their
+  current managers.
+* You may co-own fields by updating your manifest to match the existing
+  value; in this case, you'll become the manager if the other manager(s)
+  stop managing the field (remove it from their configuration).
+See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
+
+```
+
+ Notice that the field `spec.logLevel` cannot be changed using server side apply, because it is already managed by MonitoringStack `obo-demo/sample-monitoring-stack`.
+
+ Using `--force-conflicts` flag to force the change.
+
+```sh
+$ oc apply -f ./prom-spec-edited.yaml --server-side --force-conflicts 
+prometheus.monitoring.rhobs/sample-monitoring-stack serverside-applied
+```
+
+ With `--force-conflicts` flag, the field can be forced to change, but since the same field is also managed by MonitoringStack, the Observability operator
+ will detect the change, and revert it back to the value set by MonitoringStack.
+
+ Note: Some Prometheus fields generated by MonitoringStack are influenced by fields in MonitoringStack spec itself, e.g. `logLevel`. These can be changed by changing the MonitoringStack spec.
+
+ For example, to change the logLevel in Prometheus object, apply the following yaml.
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: sample-monitoring-stack
+  labels:
+    obo: example
+spec:
+  logLevel: info
+```
+
+Verify by:
+```sh
+$ oc -n obo-demo get Prometheus.monitoring.rhobs -o=jsonpath='{.items[0].spec.logLevel}' 
+info
+```
```

Comment:
We need a ## Caveats section that explains what will happen to the value already set by User if ObO decides to take ownership of in future releases

Reply:
Added a ##Caveats section
Added a Toc section